omc3/kmod_importer.py

```diff
@@ -301,6 +305,96 @@ def _get_betastar(df_model: tfs.TfsDataFrame, ip: str) -> list[float, float]:
     # return [round(bstar, 3) for bstar in df_model.loc[ip, [f"{BETA}X", f"{BETA}Y"]]]
     return df_model.loc[ip, [f"{BETA}X", f"{BETA}Y"]].tolist()
 
+def import_kmod_summary_table(
+        meas_paths: Sequence[Path | str],
+        averaged_meas_paths: Sequence[tfs.TfsDataFrame],
+        output_dir: Path | str = None
+        ) -> None:
+
+    """
+    Write the KMOD summary table from each results.tfs file.
+    It writes down the following files for each beam:
+      {beam}_kmod_sum_X.tfs: BETSTARX, ERRBETSTARX, BETWAISTX, ERRBETWAISTX, WAISTX, ERRWAISTX.
+      {beam}_kmod_sum_X.tfs: BETSTARY, ERRBETSTARY, BETWAISTY, ERRBETWAISTY, WAISTY, ERRWAISTY.
+      {beam}_tables.txt to be quickly copy/pasted in the logbook.
+    """
+
+    cols_x = ["BETSTARX", "ERRBETSTARX", "BETWAISTX", "ERRBETWAISTX", "WAISTX", "ERRWAISTX"]
+    cols_y = ["BETSTARY", "ERRBETSTARY", "BETWAISTY", "ERRBETWAISTY", "WAISTY", "ERRWAISTY"]
+    beams = [f"{BEAM_DIR}{1}", f"{BEAM_DIR}{2}"]
+
+    def output_table_single_beam(
+            data_sngl_file_raw: Path | str = None,
+            av_res_flag: bool = False,
+            ) -> tuple[tfs.TfsDataFrame, tfs.TfsDataFrame]:
+
+        """
+        It writes the single row for each kmod/results.tfs file.
+        """
+
+        if not av_res_flag:
+            file_name   = data_sngl_file_raw.parent.parent.name
+            data_sngl_file = tfs.read(data_sngl_file_raw)
+            try:
+                label = data_sngl_file["LABEL"].iloc[0]
+                second_magnet = label.split("-")[1]
+                relevant = second_magnet.split(".")[-1]
+                ip_number = relevant[1:]
+                ip_name = f"IP{ip_number}" # --- it extracts the IP from the magnet label, check for typos
+            except KeyError as e:
+                LOG.debug("Could not parse IP from LABEL. Skipping entry.", exc_info=e)
+                return None
+
+            df = data_sngl_file[cols_x + cols_y].iloc[[0]]
+            df.insert(0, "NAME", file_name)
+            df.insert(0, "IP", ip_name)
+            return df
```

Comment:
On this code branch we return `df`, just below we return `df_x, df_y`. This clashes with the return type hint of the function.

Might this be better as 2 different functions (for averaged case vs individual case) using the same helper?